### PR TITLE
Issues #1, #2, #3 & #4

### PR DESCRIPTION
**Pull request by l-lemaitre**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -100,14 +100,17 @@
...
         "phpunit/phpunit": "^9.5",
         "symfony/browser-kit": "6.2.*",
         "symfony/css-selector": "6.2.*",
         "symfony/debug-bundle": "6.2.*",
         "symfony/maker-bundle": "^1.47",
         "symfony/phpunit-bridge": "^6.1",
         "symfony/stopwatch": "6.2.*",
-        "symfony/web-profiler-bundle": "6.2.*"
+        "symfony/web-profiler-bundle": "6.2.*",
+        "theofidry/alice-data-fixtures": "^1.6"
```
Comment (IcheNouri) [inline composer.json]: à supp s'il est pas utilisé

File: config/packages/security.yaml [hunk 2]
```diff
@@ -34,15 +34,15 @@ security:
...
+            access_denied_handler: App\Security\AccessDeniedHandler
 
     # Easy way to control access for large sections of your site
     # Note: Only the *first* access control that matches will be used
     access_control:
-        - { path: ^/users/create, roles: PUBLIC_ACCESS }
         - { path: ^/login, roles: PUBLIC_ACCESS }
-        - { path: ^/admin, roles: ROLE_ADMIN }
+        - { path: ^/users/create, roles: PUBLIC_ACCESS }
+        # - { path: ^/users, roles: ROLE_ADMIN }
```
Comment (IcheNouri) [inline config/packages/security.yaml]: code commenté à supp

File: src/Controller/TaskController.php [hunk 3]
```diff
@@ -11,34 +11,60 @@
...
+
+        $routeName = $request->attributes->get('_route');
+        if ($routeName == 'app_task_list') {
+            $taskIsDone = false;
+        } else {
+            $taskIsDone = true;
+        }
+
+        return $this->render('task/list.html.twig', [
+            'tasks' => $this->doctrine->getRepository(Task::class)->findAll(),
```
Comment (IcheNouri) [inline src/Controller/TaskController.php]: filtrer dans le backend
Séparer les méthode par route ou utiliser un query param

File: src/Controller/UserController.php [hunk 4]
```diff
@@ -65,12 +86,29 @@ public function editAction(User $user, Request $request): RedirectResponse|Respo
...
     {
-        $this->userService->deleteUser($user);
+        $userConnected = $this->getUser();
+
+        $user = $this->doctrine->getRepository(User::class)->find($id);
+
+        if (!$user) {
+            throw $this->createAccessDeniedException();
+        } elseif ($user == $userConnected) {
+            $this->userService->deleteUser($user);
```
Comment (IcheNouri) [inline src/Controller/UserController.php:98]: un plus: voir si on détruit la session avant ou se déconnecter sans appeler la route logout

File: templates/task/list.html.twig [hunk 5]
```diff
@@ -3,35 +3,49 @@
...
         {% for task in tasks %}
-        <div class="col-sm-4 col-lg-4 col-md-4">
-            <div class="thumbnail">
-                <div class="caption">
-                    <h4 class="pull-right">
-                        {% if task.isDone %}<span class="glyphicon glyphicon-ok"></span>{% else %}<span class="glyphicon glyphicon-remove"></span>{% endif %}
-                    </h4>
-                    <h4><a href="{{ path('app_task_edit', {'id' : task.id }) }}">{{ task.title }}</a></h4>
-                    <p>{{ task.content }}</p>
+            {% if task.getIsDone == taskIsDone %}
```
Comment (IcheNouri) [inline templates/task/list.html.twig]: ne pas filtrer en front

File: composer.json (hunk 1 above)
Comment (l-lemaitre) [reply composer.json]: Done, see commit [2nd #1, #2, #3 & #4](https://github.com/l-lemaitre/todolist/commit/137c8a95c7faf7c70fe2a19639ad0b2256d40bb0)

File: config/packages/security.yaml (hunk 2 above)
Comment (l-lemaitre) [reply config/packages/security.yaml]: Done, see commit [2nd #1, #2, #3 & #4](https://github.com/l-lemaitre/todolist/commit/137c8a95c7faf7c70fe2a19639ad0b2256d40bb0)

File: src/Controller/TaskController.php (hunk 3 above)
Comment (l-lemaitre) [reply src/Controller/TaskController.php]: Done, see commit [2nd #1, #2, #3 & #4](https://github.com/l-lemaitre/todolist/commit/137c8a95c7faf7c70fe2a19639ad0b2256d40bb0)

File: src/Controller/UserController.php (hunk 4 above)
Comment (l-lemaitre) [reply src/Controller/UserController.php:98]: session_destroy() works, see commit [2nd #1, #2, #3 & #4](https://github.com/l-lemaitre/todolist/commit/137c8a95c7faf7c70fe2a19639ad0b2256d40bb0)

File: templates/task/list.html.twig (hunk 5 above)
Comment (l-lemaitre) [reply templates/task/list.html.twig]: Done, see commit [2nd #1, #2, #3 & #4](https://github.com/l-lemaitre/todolist/commit/137c8a95c7faf7c70fe2a19639ad0b2256d40bb0)

**Pull request merged**